nestkernel/stopwatch.h

```diff
@@ -273,18 +304,53 @@ nest::Stopwatch::print( const char* msg, timeunit_t timeunit, std::ostream& os )
 #endif
 }
 
-inline nest::Stopwatch::timestamp_t
-nest::Stopwatch::get_timestamp()
+inline nest::SingleStopwatch::timestamp_t
+nest::SingleStopwatch::get_timestamp()
 {
   // works with:
   // * hambach (Linux 2.6.32 x86_64)
   // * JuQueen (BG/Q)
   // * MacOS 10.9
   struct timeval now;
   gettimeofday( &now, static_cast< struct timezone* >( nullptr ) );
-  return ( nest::Stopwatch::timestamp_t ) now.tv_usec
-    + ( nest::Stopwatch::timestamp_t ) now.tv_sec * nest::Stopwatch::SECONDS;
+  return ( nest::SingleStopwatch::timestamp_t ) now.tv_usec
+    + ( nest::SingleStopwatch::timestamp_t ) now.tv_sec * nest::SingleStopwatch::SECONDS;
+}
+
+class StopwatchArray
+{
+public:
+  void reset();
+
+  std::vector< double > elapsed();
+
+  SingleStopwatch&
+  operator[]( const size_t tid )
```

Comment:
Do we really want to make the individual stopwatches directly available? Wouldn't it be much tidier to define `start()` and `stop()` for the array and then call `kernel().vp_manager.get_thread_id()` inside the start/stop, or use 0 if we do not use threaded timers? That would also avoid the macros, I think. The threaded-timers ifdef would then occur only inside the  `StopwatchArray` class.

Reply:
I like the idea. In the NEST code, we typically add tid as a parameter in threaded functions instead of calling `get_thread_id` every time, which would most likely be cleaner though. I can change this for this PR, but we might create a follow-up issue to do the same for all the other occurrences in the code.